compiler/rustc_mir/src/transform/const_prop.rs

```diff
@@ -393,7 +393,12 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
             .filter(|ret_layout| {
                 !ret_layout.is_zst() && ret_layout.size < Size::from_bytes(MAX_ALLOC_LIMIT)
             })
-            .map(|ret_layout| ecx.allocate(ret_layout, MemoryKind::Stack).into());
+            .and_then(|ret_layout| {
```

Comment:
Replacing `map` here by `and_then` is *not* right!
This will mean that errors turn into a `None` return place which means "function will never return", that's a huge semantic change. Please go back to `map`.

I think the right thing to do here is to simply `unwrap` the interp error...